### PR TITLE
Fix blank labels in form with repeat group with choice_filter xlsforms

### PR DESCRIPTION
**Pull request by m5r**
# Description

medic/cht-core#7550

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by m5r** [comment]
@latin-panda @garethbowen can you please review this fix please? Thanks!

**Review by garethbowen — COMMENTED** [review]
I love the e2e test!

I have a comment about a similar change but otherwise the code looks fine.

File: webapp/src/ts/services/enketo.service.ts [hunk 1]
```diff
@@ -309,7 +309,7 @@ export class EnketoService {
         const observer = new MutationObserver((mutations) => {
           const hasNewNodes = mutations.some(mutation => mutation.addedNodes.length > 0);
           if (hasFormChanged && hasNewNodes) {
-            this.currentForm.langs.setAll(language);
+            this.currentForm.langs.$formLanguages.val(language).trigger('change');
```
[the remark this answers is not in the document]
Comment (latin-panda) [reply webapp/src/ts/services/enketo.service.ts:312]: You can consider moving the set language logic (L298 - L324) to a function, so it stays together and decongest this one.

File: tests/e2e/forms/repeat-form.wdio-spec.js [hunk 2]
```diff
@@ -133,6 +134,34 @@ describe('RepeatForm', () => {
...
+
+      await (await $(kingInputPath)).click();
+      const seattleLabelPath = getCityLabelPath('seattle');
+      const redmondLabelPath = getCityLabelPath('redmond');
+      expect(await (await $(seattleLabelPath)).getText()).to.equal('Seattle');
+      expect(await (await $(redmondLabelPath)).getText()).to.equal('Redmond');
+    });
+
+    function getCityLabelPath(cityValue) {
+      return `#report-form input[name="/cascading_select/selected_city"][value="${cityValue}"] ~ .option-label.active`;
```
Comment (latin-panda) [inline tests/e2e/forms/repeat-form.wdio-spec.js:161]: you can consider expanding this one to receive the field name - `selected_city` and make it return `{ input, label }`
The you can use it for all these fields (L144, L145, L149, L150, L154 and L155) :)

**Review by latin-panda — COMMENTED** [review]
Nice job!

**Review by garethbowen — APPROVED** [review]
I appreciate the additional cleanup work, thanks!

Looks good to me. Let's get this in to AT!

**Pull request merged**